### ISSUE TITLE
Unexpected behaviour of units in Table

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#11521]
### Description

Not sure whether this is a bug, a missing feature, or just my problem:

Plain Table columns may come with a unit that is not always kept:
```python
>>> from astropy.table import Table
>>> t = Table({'foo': [1,2,3]*u.deg)
>>> print(np.copy(t['foo']))
[1. 2. 3.]
>>> print(np.copy(t['foo']*u.deg))
[1. 2. 3.] deg2
```
i.e. sometimes the column takes the unit, but sometimes not.

This is especially dangerous when there is a default unspecified unit, like sinus. Question: Does  
`np.sin(t['foo'])`take the numbers as radians or as degrees?

(I use `astropy.table.Table`, since the tables I read may have unit errors because the author added a unit even to string columns).

### Expected behavior

The unit should be consistently added (or not, if that is preferred)

### System Details

Linux-5.10.0-4-amd64-x86_64-with-glibc2.31
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]
Numpy 1.19.5
astropy 4.2
Scipy 1.6.0
Matplotlib 3.3.4

**Comment by pllim** [comment]
@taldcroft or @mhvk might know. At some point, we changed the doc to mention `QTable` over `Table` but I don't remember if this was why.

**Comment by taldcroft** [comment]
I agree the half-way support for plain Table Column with units is confusing. I have advocated for removing functionality in `Column` that makes it kind of behave like a `Quantity`. The biggest driver for plain Table `Column` with units is FITS table data where people don't really care about units and just want to use legacy (non-Quantity) code for all the manipulations. In this case the unit is really just an annotation along for the ride, not a functional part of the data.

Anyone that actually wants to use units and Quantity should use QTable, always. @pllim is correct that the Table narrative docs were updated to reflect that perspective.

**Comment by taldcroft** [comment]
What about deprecating then removing some of the quasi-unit functionality in `Column`? The most radical version is removing support for interoperability between a Column with unit and Quantity, except for initializing Quantity from a Column. Also remove `Column.to` and `Column.quantity` (which can be written `Quantity(col)`).

 @mhvk @astrofrog?

**Comment by olebole** [comment]
I would be fine with that as long as QTable runs sloppy on broken units. My initial problem was that a file couldn't be read with `QTable.read()`.

**Comment by mhvk** [comment]
I never liked `.quantity` much, as I think `Quantity(col)` is clearer and better, and can see the case for removing `.to()`, but fear we got a big stuck here, in that now whatever we do we break someone's code.... 

On `QTable`: if that couldn't read the file, that would seem a separate issue - I'd hope that it would create a Quantity with an `UnrecognizedUnit`. Or is this your other issue, #11523, about QTable trying to interpret strings as Quantity? (which is most definitely a bug!)

**Comment by olebole** [comment]
@mhvk Yes, it was #11523, and my assumption was that it is OK when some files fail to read with QTable (so I just replaced it with Table, and ran into the current issue).

**Comment by mhvk** [comment]
I don't think `QTable.read` should ever fail for something that can be read with `Table.read`...
